File: java/main/Celula.java
package main;

public class Celula {
    public int item;
    public Celula prox;

    public Celula() {
        this(Integer.MIN_VALUE, null);
    }

    public Celula(int item) {
        this.item = item;
        this.prox = null;
    }

    public Celula(int item, Celula prox) {
        this.item = item;
        this.prox = prox;
    }
}

File: c/common/celula.c
#include "celula.h"

#include <stdlib.h>

Celula * newCelula() {
    Celula * c = (Celula *) malloc(sizeof(Celula));
    c->item = 0;
    c->prox = NULL;
    return c;
}

Celula * newCelula2(int item) {
    Celula * celula = (Celula *) malloc(sizeof(Celula));

    celula->prox = NULL;
    celula->item = item;

    return celula;
}

Celula * newCelula3(Celula * prox, int item) {
    Celula * celula = (Celula *) malloc(sizeof(Celula));

    celula->prox = prox;
    celula->item = item;

    return celula;
}
